src/user.rs

```diff
@@ -88,7 +88,7 @@ pub async fn user_main(
         let thread_weighted_task =
             thread_user.weighted_tasks[weighted_bucket][weighted_bucket_position];
         let thread_task_name = &thread_task_set.tasks[thread_weighted_task].name;
-        let function = &thread_task_set.tasks[thread_weighted_task].function;
+        let function = &thread_task_set.tasks[thread_weighted_task].function.clone();
```

Comment:
This is the only real change needed for Arc - Rust could determine that due to the await something else can run during this.

Reply:
It will be interesting to see if there's any overhead from this change. I'll aim to do load testing today to compare.

Reply:
Yes - indeed. In theory this is only: atomic_inc() + atomic_dec() which compared to a .read() lock has neglect-able performance impact, but we'll see.

Reply:
no significant change in performance :+1: